operators.go

```diff
@@ -1,42 +1,53 @@
 package plugin
 
 import (
+	"fmt"
 	"net"
-	"strconv"
 
-	"github.com/corazawaf/coraza/v2"
-	"github.com/corazawaf/coraza/v2/types/variables"
-	geoip "github.com/oschwald/geoip2-golang"
+	"github.com/corazawaf/coraza/v3/collection"
+	"github.com/corazawaf/coraza/v3/experimental/plugins/plugintypes"
+	"github.com/corazawaf/coraza/v3/types/variables"
+	"github.com/oschwald/geoip2-golang"
 )
 
-type geoLookup struct{}
+// this acts as a shim for the geoip2.Reader struct so we can mock it in tests
+type geoIPReaderShim interface {
+	City(ip net.IP) (*geoip2.City, error)
+	Country(ip net.IP) (*geoip2.Country, error)
+}
 
-func (o *geoLookup) Init(data string) error {
-	return nil
+// this acts as a shim for the collection.Map struct so we can mock it in tests
+type collectionShim interface {
+	Set(key string, values []string)
 }
 
-func (o *geoLookup) Evaluate(tx *coraza.Transaction, value string) bool {
-	db, ok := tx.Waf.Config.Get("geoip", nil).(*geoip.Reader)
-	if !ok || db == nil {
-		return false
+// this acts as a shim for the plugintypes.TransactionState struct so we can mock it in tests
+type txShim interface {
+	GetGeoCollection(tx txShim) (collectionShim, error)
+}
+
+type txShimmer struct {
+	tx plugintypes.TransactionState
+}
+
```

Comment:
I believe we should add the compile time type assertion to make sure the implementation works.

```
var txShim = (*txShimmer)(nil)
```